cou/steps/plan.py

```diff
@@ -397,6 +413,26 @@ def _get_pre_upgrade_steps(analysis_result: Analysis, args: CLIargs) -> list[Pre
             )
         )
 
+    # Add a pre-upgrade step to purge old database data.
+    # This is a performance optimisation.
+    if args.purge and any(
```

Comment:
nit: For readability I think it would be nice having two sub-functions here. Something like:

```python
def _get_pre_clean_data_steps(analysis_result: Analysis, args: CLIargs) -> list[PreUpgradeStep]:
    steps = []
     if args.archive:
         steps.append(_archive_nova_cloud_controller(....))
     if args.purge:
         steps.append(_purge_nova_cloud_controller(....))
    return steps
```

Reply:
Updated